language/vm/vm_runtime/vm_runtime_types/src/native_functions/dispatch.rs

```diff
@@ -53,10 +53,7 @@ pub fn dispatch_native_function(
     NATIVE_FUNCTION_MAP.get(module)?.get(function_name)
 }
 
-macro_rules! add {
-    ($m:ident, $addr:expr, $module:expr, $name:expr, $dis:expr, $args:expr, $ret:expr) => {{
-        add!($m, $addr, $module, $name, $dis, vec![], $args, $ret)
-    }};
+macro_rules! add_with_kind {
```

Comment:
Why the separate macro versus the overloaded macro?

Reply:
Maybe `add_with_generics` or something?

Reply:
Oh, I didn't realize I could just add an extra argument. Will try that instead.